core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -73,6 +73,10 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   // The number of pending tasks which is locality required
   protected var localityAwareTasks = 0
 
+  // Executors that have been marked as disabled by the cluster manager, and are pending
+  // removal, so no tasks should be scheduled on them.
+  protected val disabledExecutors = new HashSet[String]
```

Comment:
Or maybe "executorsRemovedAndPendingLossReason", to more clearly differentiate from executorsPendingToRemove